extensions-contrib/druid-iceberg-extensions/src/main/java/org/apache/druid/iceberg/filter/IcebergRangeFilter.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.iceberg.filter;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import org.apache.iceberg.TableScan;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.expressions.Expressions;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+public class IcebergRangeFilter implements IcebergFilter
+{
+  @JsonProperty
+  private final String filterColumn;
+  @JsonProperty
+  private final Boolean lowerOpen;
+  @JsonProperty
+  private final Boolean upperOpen;
+  @JsonProperty
+  private final Object lower;
+  @JsonProperty
+  private final Object upper;
+
+  @JsonCreator
+  public IcebergRangeFilter(
+      @JsonProperty("filterColumn") String filterColumn,
+      @JsonProperty("lower") @Nullable Object lower,
+      @JsonProperty("upper") @Nullable Object upper,
+      @JsonProperty("lowerOpen") @Nullable Boolean lowerOpen,
+      @JsonProperty("upperOpen") @Nullable Boolean upperOpen
+  )
+  {
+    Preconditions.checkNotNull(filterColumn, "You must specify a filter column on the range filter");
```

Comment:
Consider using `DruidException` for these validation checks:
```suggestion
    if (filterColumn == null) {
      throw InvalidInput.exception("filterColumn cannot be null.");
    }
```

Reply:
I can refactor this in a separate PR since the other filters also need to start using DruidException

Reply:
Works for me, thanks!